From 2e620611646d7145db7026f38c68d5b4aedd5aa7 Mon Sep 17 00:00:00 2001
From: Joy Cho <joy.cho@hardkernel.com>
Date: Fri, 24 Feb 2017 16:26:22 +0900
Subject: [PATCH] ODROID-C2: Set the specific domain number to support flexible
 dvfs table to 2GHz

1. with domain '0', the max freq of dvfs table is 1.536GHz
2. to support the existing flexible dvfs table to 2.016GHz,
the domain value with setting '1' of 7th bit is reported via scpi command.

Change-Id: If39169680c79da333c3fec8b489357208eac789b
---
 drivers/amlogic/mailbox/scpi_protocol.c | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/drivers/amlogic/mailbox/scpi_protocol.c b/drivers/amlogic/mailbox/scpi_protocol.c
index 7426fa3328364..515cb327fb390 100644
--- a/drivers/amlogic/mailbox/scpi_protocol.c
+++ b/drivers/amlogic/mailbox/scpi_protocol.c
@@ -284,6 +284,14 @@ struct scpi_opp *scpi_dvfs_get_opps(u8 domain)
 	if (scpi_opps[domain])	/* data already populated */
 		return scpi_opps[domain];
 
+#if defined(CONFIG_ARCH_MESON64_ODROIDC2)
+	/*
+	 * 1. default dvfs table has max 1.536GHz with domain '0'
+	 * 2. set the 7th bit to support flexible dvfs table to 2.016GHz
+	 */
+	domain |= 0x80;
+#endif
+
 	SCPI_SETUP_DBUF(sdata, mdata, SCPI_CL_DVFS,
 			SCPI_CMD_GET_DVFS_INFO, domain, buf);
 	ret = scpi_execute_cmd(&sdata);
